libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp

```diff
@@ -274,10 +274,11 @@ void NavEKF3_core::SelectMagFusion()
         }
     }
 
-    // If we have no magnetometer and are on the ground, fuse in a synthetic heading measurement to prevent the
-    // filter covariances from becoming badly conditioned
+    // If we have no magnetometer, fuse in a synthetic heading measurement at 7Hz to prevent the filter covariances
+    // from becoming badly conditioned. For planes we only do this on-ground because they can align the yaw from GPS when
+    // airborne. For other platform types we do this all the time.
     if (!use_compass()) {
-        if (onGround && (imuSampleTime_ms - lastSynthYawTime_ms > 1000)) {
+        if ((onGround || assume_zero_sideslip()) && (imuSampleTime_ms - lastSynthYawTime_ms > 140)) {
```

Comment:
According to the comment I think this should be:

`onGround || !assume_zero_sideslip()`

Either we are on the ground and do it for all vehicles or we aren't and we do it for non-plane vehicles.